llvm/lib/Transforms/Scalar/DeadStoreElimination.cpp

```diff
@@ -1602,7 +1731,17 @@ struct DSEState {
 
       // Uses which may read the original MemoryDef mean we cannot eliminate the
       // original MD. Stop walk.
-      if (isReadClobber(MaybeDeadLoc, UseInst)) {
+      // If KillingDef is a CallInst with "initializes" attribute, the reads in
+      // Callee would be dominated by initializations, so this should be safe.
+      bool IsKillingDefFromInitAttr = false;
+      if (IsInitializesAttrMemLoc) {
+        if (KillingI == UseInst &&
+            KillingUndObj == getUnderlyingObject(MaybeDeadLoc.Ptr)) {
```

Comment:
It looks like this inner if is untested (test pass if I replace the condition with true).

Reply:
Nice catch! Added a new unit test, `p1_write_then_read_caller_with_clobber`, to test this inner condition.